test/AutoDiff/array.swift

```diff
@@ -24,20 +25,47 @@ ArrayAutoDiffTests.test("ArraySubscript") {
   }
 
   expectEqual(
-    FloatArrayGrad([1, 1, 1, 0, 0, 0]),
+    FloatArrayTan([1, 1, 1, 0, 0, 0]),
     gradient(at: [2, 3, 4, 5, 6, 7], in: sumFirstThree))
 }
 
 ArrayAutoDiffTests.testWithLeakChecking("ArrayLiteral") {
-  func twoElementLiteral(_ x: Tracked<Float>, _ y: Tracked<Float>) -> [Tracked<Float>] {
-    return [x, y]
+  do {
+    func twoElementLiteral(_ x: Tracked<Float>, _ y: Tracked<Float>) -> [Tracked<Float>] {
+      return [x, y]
+    }
+    let pb = pullback(at: 1, 1, in: twoElementLiteral)
+    let (gradX, gradY) = pb(TrackedFloatArrayTan([Tracked<Float>(1), Tracked<Float>(2)]))
+    expectEqual(1, gradX)
+    expectEqual(2, gradY)
   }
 
-  let (gradX, gradY) = pullback(at: Tracked<Float>(1), Tracked<Float>(1), in: twoElementLiteral)(
-    Array<Tracked<Float>>.DifferentiableView([Tracked<Float>(1), Tracked<Float>(2)]))
+  do {
+    // TF-952: Test array literal initialized from an address (e.g. `var`).
+    func twoElementLiteralAddress(_ x: Tracked<Float>, _ y: Tracked<Float>) -> [Tracked<Float>] {
+      var result = x
+      result = result * y
+      return [result, result]
+    }
+    let pb = pullback(at: 3, 4, in: twoElementLiteralAddress)
+    let (gradX, gradY) = pb(TrackedFloatArrayTan([1, 1]))
+    expectEqual(8, gradX)
+    expectEqual(6, gradY)
+  }
 
-  expectEqual(Tracked<Float>(1), gradX)
-  expectEqual(Tracked<Float>(2), gradY)
+  do {
+    // TF-952: Test array literal initialized with function call results.
+    func twoElementLiteralFunctionResult(_ x: Tracked<Float>, _ y: Tracked<Float>) -> [Tracked<Float>] {
```

Comment:
Note: this `twoElementLiteralFunctionResult` test continues to fail with this patch.
Note that the activity info (for `testArrayUninitializedIntrinsicFunctionResult` in `activity_analysis.swift`) is unchanged by this patch.